BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -2006,7 +2006,7 @@ void BeamRideGuidance()
                 guidanceActive = false;
                 return;
             }
-            Ray laserBeam = new Ray(targetingPod.cameraParentTransform.position + (targetingPod.vessel.Velocity() * Time.fixedDeltaTime), targetingPod.targetPointPosition - targetingPod.cameraParentTransform.position);
+            Ray laserBeam = new Ray(targetingPod.cameraParentTransform.position + targetingPod.cameraParentTransform.right*0.25f + targetingPod.cameraParentTransform.up*0.2f + (targetingPod.vessel.Velocity() * Time.fixedDeltaTime), targetingPod.targetPointPosition - targetingPod.cameraParentTransform.position);
```

Comment:
What happens if the missile approaches the beam from the lower left or is particularly fat? Won't it still get locked onto in those cases?
I think a better solution is to simply not update `targetPointPosition` in `ModuleTargetingCamera.GetHitPoint` if the raycast hits a missile launched by the craft. Moving the `rayHit.collider` → hit part code to the start of that block and returning early should be sufficient.

Reply:
I suspect that the offset you're trying to do is already handled by ``beamCorrectionFactor``, which looks to be adding an offset to a beamriding missile (if present in the missile.cfg). 
Echoing Doc, could do something as simple as 

```
KerbalEVA hitEVA = rayHit.collider.gameObject.GetComponentUpwards<KerbalEVA>();
Part p = hitEVA ? hitEVA.part : rayHit.collider.GetComponentInParent<Part>();
if (p.GetComponent<MissileLauncher>()) return;
groundStabilized = true;
groundTargetPosition = rayHit.point;

 if (CoMLock)
{
      if (p && p.vessel && p.vessel.CoM != Vector3.zero)
 ... 
```
for ``GroundStabilize()`` and ``GetHitPoint()``

Reply:
`beamCorrectionFactor` should just be the P for the PID of the missile as far as I know. As both of your suggestions, yes that's basically what I did. I did have to move the `p && p.vessel` check up though to ensure that it actually exists first since I'm getting the SourceVessel. I'm sure there's going to be use cases where you do want to target an enemy missile with a SALH weapon so I've gone and done that extra step so we get:
```
KerbalEVA hitEVA = rayHit.collider.gameObject.GetComponentUpwards<KerbalEVA>();
Part p = hitEVA ? hitEVA.part : rayHit.collider.GetComponentInParent<Part>();

bool pCheck = false;

if (p && p.vessel)
{
    var pMissile = p.FindModuleImplementing<MissileLauncher>();
    if (pMissile != null)
    {
        if (pMissile.SourceVessel == vessel) return;
    }
    pCheck = true;
}
``` instead where pCheck is just there to prevent having to assess that again.